### PR TITLE
Fleshed out guild join page

### LINKED ISSUE #896: Standardize guild onboarding copy
### What would you like to be added?

We currently have [this Notion page](https://meta-game.notion.site/Guild-Onboarding-Why-How-e3d7383d86de4f7f8240a20cd92b8efb) discussing Guild Onboarding. A lot of this content should probably be in our _actual_ guild onboarding page in [MyMeta](https://my.metagame.wtf/join/guild). This will require some design work and copy work.

### Why is this needed?

Because potential new guilds should have more context about joining: why, if they're a good fit, etc.

### PR DESCRIPTION
**Pull request by alalonde**
## Overview

**What features/fixes does this PR include?**
Fleshes out the guild join page with additional copy that can help the applicant determine fit.
Closes #896 

## Follow up Improvement Ideas
- Seems that guilds should be able to join MetaGame without linking Discord along the way. We would need an alternative flow which basically goes right to the form with possibly the DAO address required so we can pull memberships from DaoHaus
- This page still hasn't been properly designed :D

## Assets
image

### CODE REVIEW

**Comment by alalonde** [comment]
Mobile version is no bueno currently, should be some simple tweaks to make it better.

File: packages/web/components/Guild/GuildJoin.tsx [hunk 1]
```diff
@@ -105,3 +87,238 @@ export const GuildJoin: React.FC = () => {
...
+      </UnorderedList>
+      <Text fontWeight="bold">Read more about MetaGame:</Text>
+      <UnorderedList>
+        <ListItem>
+          <MetaLink href="/">MetaGame</MetaLink>
+        </ListItem>
+        <ListItem>
+          <MetaLink
+            isExternal
+            href="https://wiki.metagame.wtf/docs/wtf-is-metagame/wtf-is-metagame"
```
Comment (vidvidvid) [inline packages/web/components/Guild/GuildJoin.tsx:168]: maybe you could link them to `/learn/wiki`

Comment (alalonde) [reply packages/web/components/Guild/GuildJoin.tsx:173]: ah, good idea

**Comment by mquellhorst** [comment]
Two things:

- I notice that when I'm not logged in to Notion I'm unable to access the manifesto. Maybe we can direct somewhere else or move that page from Notion?
- Any thoughts on how we link to this page from the Guilds overview page or from the menu (Contribute as Guild)?

**Review by mquellhorst — APPROVED** [review]
The access of the Notion page should be changed so anyone can see it without the need to login

**Pull request merged**